src/rust/engine/fs/src/lib.rs

```diff
@@ -615,14 +615,17 @@ impl PosixFS {
     })
   }
 
-  pub fn scandir(
-    &self,
+  ///
+  /// TODO: See the note on references in ASYNC.md.
+  ///
+  pub fn scandir<'a, 'b>(
+    &'a self,
     dir_relative_to_root: Dir,
-  ) -> impl Future<Item = DirectoryListing, Error = io::Error> {
+  ) -> BoxFuture<'b, Result<DirectoryListing, io::Error>> {
     let vfs = self.clone();
-    self.executor.spawn_on_io_pool(future::lazy(move || {
-      vfs.scandir_sync(&dir_relative_to_root)
-    }))
+    self
+      .executor
+      .spawn_blocking(move || vfs.scandir_sync(&dir_relative_to_root))
```

Comment:
It will be interesting to see if this regresses perf in the way that when I tried to use the blocking stuff last time it did... 🤞

Reply:
`spawn_blocking` is a different API, afaict. It's basically: "please literally dedicate me a thread". So it should be a 1:1 replacement for the `CpuPool` we used to have here.

The `tokio-fs` blocking APIs are definitely still a thing, but having removed one moving piece is nice.